libpod/options.go

```diff
@@ -2342,3 +2423,64 @@ func WithVolatile() CtrCreateOption {
 		return nil
 	}
 }
+
+// WithCpusCtr computes the CPU period and quota given the CPU core count for a pod
+func WithCPUs(num float64) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+		if num > float64(runtime.NumCPU()) {
+			return errors.New("Invalid CPU amount, exceeds total number available")
+		}
+
+		pod.config.CPUs = num
+		return nil
+	}
+}
+
+// WithCpuSetCpusCtr computes and sets the Cpus linux resource string which determines the amount of cores, from those available,  we are allowed to execute on
+func WithCPUSetCPUs(inp string) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+		re := regexp.MustCompile(`\d+`)
+		match := re.FindAllString(inp, -1)
+		for i := 0; i < len(match); i++ {
+			val, err := strconv.ParseFloat(match[i], 64)
+			if err != nil {
+				errors.Wrapf(err, "failed to parse float from CPU values")
+			}
+			if val > float64(runtime.NumCPU()) {
+				inp = "0-" + fmt.Sprint(runtime.NumCPU())
+			}
+		}
+		/*
+			wholeNum := ""
+			for j := 0; j < len(inp); j++ {
+				temp := inp[j]
+				tempS := string(temp)
+				if len(wholeNum) == 0 || len(wholeNum) == 1 {
+					wholeNum = tempS
+				}
+				if val, err := strconv.Atoi(wholeNum); err == nil {
+					if int(runtime.NumCPU()) < val {
+						max := fmt.Sprint(runtime.NumCPU())
+						totalNew := "0-" + max
+						inp = totalNew
+					}
+				}
+				if j != len(inp)-1 {
+					if (string(inp[j+1]) != "-") && (string(inp[j+1]) != ",") && wholeNum != "-" && wholeNum != "," {
+						wholeNum += string(inp[j+1])
+					}
+				}
+			}
+		*/
```

Comment:
If not needed, yank this or add a todo.

Reply:
Will do, I think @jwhonce wanted to take a look at my different implementations so I left them in for a bit.